module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1539,29 +1553,48 @@ public function finesAction()
         $result = $catalog->getMyFines($patron);
         $fines = [];
         $totalDue = 0;
-        foreach ($result as $row) {
-            // Attempt to look up and inject title:
+        $driversNeeded = [];
+        foreach ($result as $i => $row) {
+            // If we have an id, add it to list of record drivers to load:
+            if ($row['id'] ?? false) {
+                $driversNeeded[$i] = [
+                    'id' => $row['id'],
+                    'source' => $row['source'] ?? DEFAULT_SEARCH_BACKEND
+                ];
+            }
+            $totalDue += $row['balance'] ?? 0;
+            // Store by original index so that we can access it when loading record
+            // drivers:
+            $fines[$i] = $row;
+        }
+
+        if ($driversNeeded) {
+            $recordLoader = $this->serviceLocator->get(\VuFind\Record\Loader::class);
             try {
-                if (strlen($row['id'] ?? '') > 0) {
-                    $source = $row['source'] ?? DEFAULT_SEARCH_BACKEND;
-                    $row['driver'] = $this->serviceLocator
-                        ->get(\VuFind\Record\Loader::class)
-                        ->load($row['id'], $source);
-                    if (empty($row['title'])) {
-                        $row['title'] = $row['driver']->getShortTitle();
+                $drivers = $recordLoader->loadBatch($driversNeeded);
```

Comment:
Could you just use the `$tolerateBackendExceptions` flag of the `loadBatch()` method here to eliminate the need for the record-by-record fallback, or is there a different specific case that is intended to protect against?

Reply:
The old code tolerates any kind of exception, so I tried to follow that here as well. Our custom version of the old getDriverForILSRecord actually has been added to make any kind of exception tolerable. Do you think it would make sense to incorporate support for that in Loader's load/loadBatch?

Reply:
I guess the question is whether there's any kind of exception other than a backend exception that would occur inside loadBatch() that would result in some records loading but others being skipped. If there's a realistic, likely scenario that we need to work around, then I'm certainly open to making the record loader more flexible -- perhaps we could replace $tolerateBackendException with a $toleranceLevel parameter that could accept a boolean (for backward compatibility with existing behavior) or a flag constant (to add different behaviors). But if the existing behavior covers the scenario where we would need to try again with one-by-one loading, and any other exception just indicates that we can't load records at all, then I would be in favor of simplifying the code. Either way, I think it would be a good idea to use the logger to report on any exceptions we encounter, to ensure that we don't accidentally mask problems completely.

Reply:
You're right, there should only be backend exceptions from the backend, and I verified that with Solr in particular. I simplified the code to just use loadBatch.

Reply:
Thanks!